downstairs/src/extent_inner_raw.rs

```diff
@@ -325,24 +325,42 @@ impl ExtentInner for RawInner {
         // operation atomic.
         self.set_flush_number(new_flush, new_gen)?;
 
-        // Now, we fsync to ensure data is flushed to disk.  It's okay to crash
-        // before this point, because setting the flush number is atomic.
-        cdt::extent__flush__file__start!(|| {
-            (job_id.get(), self.extent_number, 0)
-        });
-        if let Err(e) = self.file.sync_all() {
-            /*
-             * XXX Retry?  Mark extent as broken?
-             */
-            return Err(CrucibleError::IoError(format!(
-                "extent {}: fsync 1 failure: {e:?}",
-                self.extent_number,
-            )));
+        if cfg!(feature = "omicron-build") {
+            // no-op, FIOFFS will be called in region_flush
+        } else {
+            // Now, we fsync to ensure data is flushed to disk.  It's okay to crash
+            // before this point, because setting the flush number is atomic.
+            cdt::extent__flush__file__start!(|| {
+                (job_id.get(), self.extent_number, 0)
+            });
+
+            if let Err(e) = self.file.sync_all() {
+                /*
+                 * XXX Retry?  Mark extent as broken?
+                 */
+                return Err(CrucibleError::IoError(format!(
+                    "extent {}: fsync 1 failure: {e:?}",
+                    self.extent_number,
+                )));
+            }
+
+            cdt::extent__flush__file__done!(|| {
+                (job_id.get(), self.extent_number, 0)
+            });
         }
+
+        cdt::extent__flush__done!(|| { (job_id.get(), self.extent_number, 0) });
```

Comment:
I feel like this should be at the beginning of `post_flush`, so that it accounts for the actual flush time (even though that's amortized across many extents)